test/ng/directive/ngOptionsSpec.js

```diff
@@ -2217,6 +2217,21 @@ describe('ngOptions', function() {
       dealoc(element);
     }));
 
+    // See issue #13331
+    it('should set [selected] to false when a genuine option is selected', function() {
+      scope.options = ['a'];
+      scope.model = '';
+      compile(
+          '<select ng-model="model" ng-options="opt for opt in options">' +
+            '<option id="empty_option" value="">Empty Option</option>' +
+          '</select>');
+      var opt = element.find("option")[0];
+      expect(opt.getAttribute("selected")).toBeTruthy();
+      scope.model = 'a';
+      scope.$digest();
+      expect(opt.getAttribute("selected")).toBeFalsy();
```

Comment:
I think you are right about using `hasAttribute` here since `<option value="value2" selected>Value 2</option>` is a valid selected option. Elsewhere in this spec we just look at the `opt.selected` property instead. Is there some reason why we are checking the attribute rather than the property here?